backend/main.py

```diff
@@ -114,14 +114,23 @@ async def start_spotihue(lights: List[str] = None):
     if not lights:
         raise HTTPException(status_code=400, detail='"lights" list is required.')
 
-    available_lights = spotihue.retrieve_available_lights()
+    available_lights = [light['light_name'] for light in spotihue.retrieve_available_lights()]
     lights = [light for light in lights if light in available_lights]
 
     try:
-        # TODO: make this idempotent
+        spotihue_task_running = tasks.is_spotihue_running()
 
-        task = tasks.run_spotihue.delay(lights)
-        redis_client.set("spotihue", str(task.id))
+        if spotihue_task_running:
+            logger.info('Spotihue is already running')
```

Comment:
In a future PR, we could consider taking the diff between currently running task lights and new request's input lights, and using task.replace({new lights list}) in the case of the lights lists differing (ie, replacing the currently running task in 1 endpoint call instead of terminating it and then re-running it in 2 endpoint calls)

Reply:
I like this idea.

Reply:
Capturing it in an issue